core/genesis.go

```diff
@@ -246,21 +246,115 @@ func SetupGenesisBlockWithOverride(db ethdb.Database, genesis *Genesis, override
 	return newcfg, stored, nil
 }
 
+// Hard fork block height specified in config.toml has higher priority, but
+// if it is not specified in config.toml, use the default height in code.
 func (g *Genesis) configOrDefault(ghash common.Hash) *params.ChainConfig {
+	var defaultConfig *params.ChainConfig
 	switch {
-	case g != nil:
-		return g.Config
 	case ghash == params.MainnetGenesisHash:
-		return params.MainnetChainConfig
+		defaultConfig = params.MainnetChainConfig
 	case ghash == params.BSCGenesisHash:
-		return params.BSCChainConfig
+		defaultConfig = params.BSCChainConfig
```

Comment:
this give a chance to wrong config in bsc mainnet and chapelnet
so when match BSCGenesisHash or ChapelGenesisHash, just return as before?